### PR TITLE
Add spellchecker to the docs

### PR DESCRIPTION
**Pull request by kaedroho**
This PR adds a new make target for the docs called ``make spelling`` and a wordlist. This checks through the docs and reports any spelling errors. I've also made a start at fixing the spelling errors that currently exist in Wagtails docs.

Shall we write the docs in British or American English? (currently, we use a mix of the two).

### CODE REVIEW

File: docs/releases/1.0.rst [hunk 1]
```diff
@@ -84,9 +84,9 @@ Admin
  * Removed the dependency on ``LOGIN_URL`` and ``LOGIN_REDIRECT_URL`` settings
  * Password reset view names namespaced to wagtailadmin
  * Removed the need to add permission check on admin views (now automated)
- * Reversing ``django.contrib.auth.admin.login`` will no longer lead to Wagtails login view (making it easier to have front end views)
+ * Reversing ``django.contrib.auth.admin.login`` will no longer lead to Wagtails log in view (making it easier to have frontend log in views)
  * Added cache-control headers to all admin views. This allows Varnish/Squid/CDN to run on vanilla settings in front of a Wagtail site
- * Date / time pickers now consistently use times without seconds, to prevent Javascript behaviour glitches when focusing / unfocusing fields
+ * Date / time pickers now consistently use times without seconds, to prevent JavasSript behaviour glitches when focusing / unfocusing fields
```
Comment (tomdyson) [inline docs/releases/1.0.rst]: JavasSript -> JavaScript

**Comment by tomdyson** [comment]
I vote British!

File: docs/releases/1.0.rst [hunk 2]
```diff
@@ -84,9 +84,9 @@ Admin
  * Removed the dependency on ``LOGIN_URL`` and ``LOGIN_REDIRECT_URL`` settings
  * Password reset view names namespaced to wagtailadmin
  * Removed the need to add permission check on admin views (now automated)
- * Reversing ``django.contrib.auth.admin.login`` will no longer lead to Wagtails login view (making it easier to have front end views)
+ * Reversing ``django.contrib.auth.admin.login`` will no longer lead to Wagtails log in view (making it easier to have frontend log in views)
```
Comment (zerolab) [inline docs/releases/1.0.rst]: Wagtails -> Wagtail's
In this particular case, should be "Wagtail's login view" as that is not an action.

File: docs/releases/1.0.rst [hunk 3]
```diff
@@ -219,15 +219,15 @@ should work fine without them.
 Login/Password reset views renamed
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-It was previously possible to reverse the Wagtail login using django.contrib.auth.views.login.
+It was previously possible to reverse the Wagtail log in view using ``django.contrib.auth.views.login``.
```
Comment (zerolab) [inline docs/releases/1.0.rst]: Same as above "login view"

**Comment by zerolab** [comment]
:+1:

Also voting fot British.

 *goes to install a British spellchecker*

**Comment by kaedroho** [comment]
Please merge if happy. As there are so many spelling mistakes at the moment, I think it would be best to tackle it one section at a time.

**Comment by tomdyson** [comment]
Merged. Great stuff @kaedroho!

**Comment by gasman** [comment]
Does `sphinxcontrib-spelling` need to be added to docs/requirements.txt as well as /requirements-dev.txt? Presumably it does, if the former is what readthedocs is using.

**Comment by kaedroho** [comment]
Ahh, it might do actually (RTD doesn't run the spellcheck though)

**Comment by gasman** [comment]
Certainly `make html` fails without that package installed.

Also, having installed it locally, I now see that I also need the 'enchant' C library. Perhaps this calls for some actual developer setup docs...

**Pull request merged**